tensorflow/core/kernels/mkl_input_conversion_op.cc

```diff
@@ -136,19 +143,32 @@ class MklInputConversionOp : public OpKernel {
                                     input_tensor_0.shape(),
                                     mkl_output_mkl_shape);
 
-          // Create MklDnnData object for input0 tesnsor
-          auto cpu_engine = engine(engine::cpu, 0);
+          // Create MklDnnData object for input0 tensor
+          auto cpu_engine = engine(ENGINE_CPU, 0);
           MklDnnData<T> input(&cpu_engine);
           input.SetUsrMem(input0_md, &input_tensor_0);
-
           // Create reorder from input0's layout to input1's layout
           std::vector<primitive> net;
-          CHECK_EQ(input.CheckReorderToOpMem(
-                       memory::primitive_desc(input1_md, cpu_engine),
-                       tensor_out, &net),
-                   true);
-          stream(stream::kind::eager).submit(net).wait();
-
+#ifdef ENABLE_MKLDNN_V1
+          std::vector<MemoryArgsMap> net_args;
+          // TODO(bhavanis): Refactor CheckReorderToOpMem() to create and
+          // execute reorder
+          auto status = input.CheckReorderToOpMem(input1_md, tensor_out, net,
+                                                  net_args, cpu_engine);
+          OP_REQUIRES(
+              context, status,
+              errors::Internal(
+                  "MklInputConversionOp: Failed to create reorder for input0"));
+          ExecutePrimitive(net, &net_args, cpu_engine);
```

Comment:
Thank you for refactoring `ExecutePrimitive`! Let's `#define NET_ARGS_PTR` (to `&new_args` vs `nullptr`) so we can put `ExecutePrimitive` calls outside of `#ifdef`. `OP_REQUIRES` can be pulled off of `#ifdef` as well. 

Refactoring `CheckReorderToOpMem` sounds good. Thank you for adding the TODO! :)

Reply:
Thanks @penpornk! Defined another macro per your suggestion.